cluster-provision/images/vm-image-builder/create-containerdisk.sh

```diff
@@ -6,90 +6,105 @@ if [ "$#" -ne 1 ]; then
 
 export CUSTOMIZE_IMAGE_SCRIPT=${CUSTOMIZE_IMAGE_SCRIPT:-"${SCRIPT_PATH}/customize-image.sh"}
 
 function customize_image() {
-  local source_image=$1
-  local os_variant=$2
-  local customized_image=$3
-  local cloud_config=$4
+    local source_image=$1
+    local os_variant=$2
+    local customized_image=$3
+    local cloud_config=$4
 
-  # Backup the VM image and pass copy of the original image
-  # in case customizing script fail.
-  vm_image_copy="$(dirname "${customized_image}")/copy-${source_image}"
-  cp "${source_image}" "${vm_image_copy}"
+    # Backup the VM image and pass copy of the original image
+    # in case customizing script fail.
+    vm_image_copy="$(dirname "${customized_image}")/copy-${source_image}"
+    cp "${source_image}" "${vm_image_copy}"
 
-  # TODO: convert this script and its dependencies to container
-  ${CUSTOMIZE_IMAGE_SCRIPT} "${vm_image_copy}" "${os_variant}" "${customized_image}" "${cloud_config}"
+    # TODO: convert this script and its dependencies to container
+    ${CUSTOMIZE_IMAGE_SCRIPT} "${vm_image_copy}" "${os_variant}" "${customized_image}" "${cloud_config}"
 
-  # Backup no longer needed.
-  rm -f "${vm_image_copy}"
+    # Backup no longer needed.
+    rm -f "${vm_image_copy}"
 }
 
 function cleanup() {
-  if [ $? -ne 0 ]; then
-    rm -rf "${build_directory}"
-  fi
+    if [ $? -ne 0 ]; then
+        rm -rf "${build_directory}"
+    fi
 
-  rm -f "copy-${VM_IMAGE}"
+    rm -f "copy-${VM_IMAGE}"
 }
 
+trap 'cleanup' EXIT SIGINT
+
 export IMAGE_NAME=$1
 
 if [ -f "${SCRIPT_PATH}/${IMAGE_NAME}/create-image.sh" ]; then
     export TAG="devel"
     readonly VM_IMAGE="provisioned-image.qcow2"
     readonly build_directory="${SCRIPT_PATH}/${IMAGE_NAME}/build"
-    
+
     trap 'cleanup' EXIT SIGINT
-    
+
     mkdir "${build_directory}"
 
     pushd "${SCRIPT_PATH}/${IMAGE_NAME}"
       cleanup
       echo "Creating the image"
-      ./create-image.sh "${build_directory}/${VM_IMAGE}" 
+      ./create-image.sh "${build_directory}/${VM_IMAGE}"
 
       echo "Creating the containerdisk ..."
       docker build . -t ${IMAGE_NAME}:${TAG} -f - <<END
 FROM scratch
 ADD --chown=107:107 build/${VM_IMAGE} /disk/
 END
     popd
-
-else 
-    export TAG=devel
+else
     export OS_VARIANT="$(cat ${SCRIPT_PATH}/${IMAGE_NAME}/os-variant)"
     export CLOUD_CONFIG_PATH="${SCRIPT_PATH}/${IMAGE_NAME}/cloud-config"
-    export VM_IMAGE_URL="$(cat ${SCRIPT_PATH}/${IMAGE_NAME}/image-url)"
-
-    readonly VM_IMAGE="source-image.qcow2"
-    readonly build_directory="${IMAGE_NAME}_build"
-    readonly new_vm_image_name="provisioned-image.qcow2"
 
+    if [[ ${ARCH} = "aarch64" ]]; then
+        export VM_IMAGE_URL="$(cat ${SCRIPT_PATH}/${IMAGE_NAME}/image-url-aarch64)"
+        export TAG=devel-aarch64
+        readonly VM_IMAGE="source-image-aarch64.qcow2"
+        readonly build_directory="${IMAGE_NAME}_build_aarch64"
+        readonly new_vm_image_name="provisioned-image-aarch64.qcow2"
```

Comment:
Please consider minimizing the arch footprint and change only when needed
`build_directory`,`new_vm_image_name` and `VM_IMAGE` can be left as they are.
Note that what matter is the name of the artifact which is the container image name and tag.